// Code generated by protoc-gen-validate. DO NOT EDIT.
// source: v1/order.proto

package orderv1

import (
	"bytes"
	"errors"
	"fmt"
	"net"
	"net/mail"
	"net/url"
	"regexp"
	"sort"
	"strings"
	"time"
	"unicode/utf8"

	"google.golang.org/protobuf/types/known/anypb"
)

// ensure the imports are used
var (
	_ = bytes.MinRead
	_ = errors.New("")
	_ = fmt.Print
	_ = utf8.UTFMax
	_ = (*regexp.Regexp)(nil)
	_ = (*strings.Reader)(nil)
	_ = net.IPv4len
	_ = time.Duration(0)
	_ = (*url.URL)(nil)
	_ = (*mail.Address)(nil)
	_ = anypb.Any{}
	_ = sort.Sort
)

// define the regex for a UUID once up-front
var _order_uuidPattern = regexp.MustCompile("^[0-9a-fA-F]{8}-[0-9a-fA-F]{4}-[0-9a-fA-F]{4}-[0-9a-fA-F]{4}-[0-9a-fA-F]{12}$")

// Validate checks the field values on PlaceOrderReq with the rules defined in
// the proto definition for this message. If any rules are violated, the first
// error encountered is returned, or nil if there are no violations.
func (m *PlaceOrderReq) Validate() error {
	return m.validate(false)
}

// ValidateAll checks the field values on PlaceOrderReq with the rules defined
// in the proto definition for this message. If any rules are violated, the
// result is a list of violation errors wrapped in PlaceOrderReqMultiError, or
// nil if none found.
func (m *PlaceOrderReq) ValidateAll() error {
	return m.validate(true)
}

func (m *PlaceOrderReq) validate(all bool) error {
	if m == nil {
		return nil
	}

	var errors []error

	if utf8.RuneCountInString(m.GetCurrency()) != 3 {
		err := PlaceOrderReqValidationError{
			field:  "Currency",
			reason: "value length must be 3 runes",
		}
		if !all {
			return err
		}
		errors = append(errors, err)

	}

	if all {
		switch v := interface{}(m.GetAddress()).(type) {
		case interface{ ValidateAll() error }:
			if err := v.ValidateAll(); err != nil {
				errors = append(errors, PlaceOrderReqValidationError{
					field:  "Address",
					reason: "embedded message failed validation",
					cause:  err,
				})
			}
		case interface{ Validate() error }:
			if err := v.Validate(); err != nil {
				errors = append(errors, PlaceOrderReqValidationError{
					field:  "Address",
					reason: "embedded message failed validation",
					cause:  err,
				})
			}
		}
	} else if v, ok := interface{}(m.GetAddress()).(interface{ Validate() error }); ok {
		if err := v.Validate(); err != nil {
			return PlaceOrderReqValidationError{
				field:  "Address",
				reason: "embedded message failed validation",
				cause:  err,
			}
		}
	}

	// no validation rules for Email

	for idx, item := range m.GetOrderItems() {
		_, _ = idx, item

		if all {
			switch v := interface{}(item).(type) {
			case interface{ ValidateAll() error }:
				if err := v.ValidateAll(); err != nil {
					errors = append(errors, PlaceOrderReqValidationError{
						field:  fmt.Sprintf("OrderItems[%v]", idx),
						reason: "embedded message failed validation",
						cause:  err,
					})
				}
			case interface{ Validate() error }:
				if err := v.Validate(); err != nil {
					errors = append(errors, PlaceOrderReqValidationError{
						field:  fmt.Sprintf("OrderItems[%v]", idx),
						reason: "embedded message failed validation",
						cause:  err,
					})
				}
			}
		} else if v, ok := interface{}(item).(interface{ Validate() error }); ok {
			if err := v.Validate(); err != nil {
				return PlaceOrderReqValidationError{
					field:  fmt.Sprintf("OrderItems[%v]", idx),
					reason: "embedded message failed validation",
					cause:  err,
				}
			}
		}

	}

	if len(errors) > 0 {
		return PlaceOrderReqMultiError(errors)
	}

	return nil
}

// PlaceOrderReqMultiError is an error wrapping multiple validation errors
// returned by PlaceOrderReq.ValidateAll() if the designated constraints
// aren't met.
type PlaceOrderReqMultiError []error

// Error returns a concatenation of all the error messages it wraps.
func (m PlaceOrderReqMultiError) Error() string {
	msgs := make([]string, 0, len(m))
	for _, err := range m {
		msgs = append(msgs, err.Error())
	}
	return strings.Join(msgs, "; ")
}

// AllErrors returns a list of validation violation errors.
func (m PlaceOrderReqMultiError) AllErrors() []error { return m }

// PlaceOrderReqValidationError is the validation error returned by
// PlaceOrderReq.Validate if the designated constraints aren't met.
type PlaceOrderReqValidationError struct {
	field  string
	reason string
	cause  error
	key    bool
}

// Field function returns field value.
func (e PlaceOrderReqValidationError) Field() string { return e.field }

// Reason function returns reason value.
func (e PlaceOrderReqValidationError) Reason() string { return e.reason }

// Cause function returns cause value.
func (e PlaceOrderReqValidationError) Cause() error { return e.cause }

// Key function returns key value.
func (e PlaceOrderReqValidationError) Key() bool { return e.key }

// ErrorName returns error name.
func (e PlaceOrderReqValidationError) ErrorName() string { return "PlaceOrderReqValidationError" }

// Error satisfies the builtin error interface
func (e PlaceOrderReqValidationError) Error() string {
	cause := ""
	if e.cause != nil {
		cause = fmt.Sprintf(" | caused by: %v", e.cause)
	}

	key := ""
	if e.key {
		key = "key for "
	}

	return fmt.Sprintf(
		"invalid %sPlaceOrderReq.%s: %s%s",
		key,
		e.field,
		e.reason,
		cause)
}

var _ error = PlaceOrderReqValidationError{}

var _ interface {
	Field() string
	Reason() string
	Key() bool
	Cause() error
	ErrorName() string
} = PlaceOrderReqValidationError{}

// Validate checks the field values on OrderItem with the rules defined in the
// proto definition for this message. If any rules are violated, the first
// error encountered is returned, or nil if there are no violations.
func (m *OrderItem) Validate() error {
	return m.validate(false)
}

// ValidateAll checks the field values on OrderItem with the rules defined in
// the proto definition for this message. If any rules are violated, the
// result is a list of violation errors wrapped in OrderItemMultiError, or nil
// if none found.
func (m *OrderItem) ValidateAll() error {
	return m.validate(true)
}

func (m *OrderItem) validate(all bool) error {
	if m == nil {
		return nil
	}

	var errors []error

	if all {
		switch v := interface{}(m.GetItem()).(type) {
		case interface{ ValidateAll() error }:
			if err := v.ValidateAll(); err != nil {
				errors = append(errors, OrderItemValidationError{
					field:  "Item",
					reason: "embedded message failed validation",
					cause:  err,
				})
			}
		case interface{ Validate() error }:
			if err := v.Validate(); err != nil {
				errors = append(errors, OrderItemValidationError{
					field:  "Item",
					reason: "embedded message failed validation",
					cause:  err,
				})
			}
		}
	} else if v, ok := interface{}(m.GetItem()).(interface{ Validate() error }); ok {
		if err := v.Validate(); err != nil {
			return OrderItemValidationError{
				field:  "Item",
				reason: "embedded message failed validation",
				cause:  err,
			}
		}
	}

	// no validation rules for Cost

	if len(errors) > 0 {
		return OrderItemMultiError(errors)
	}

	return nil
}

// OrderItemMultiError is an error wrapping multiple validation errors returned
// by OrderItem.ValidateAll() if the designated constraints aren't met.
type OrderItemMultiError []error

// Error returns a concatenation of all the error messages it wraps.
func (m OrderItemMultiError) Error() string {
	msgs := make([]string, 0, len(m))
	for _, err := range m {
		msgs = append(msgs, err.Error())
	}
	return strings.Join(msgs, "; ")
}

// AllErrors returns a list of validation violation errors.
func (m OrderItemMultiError) AllErrors() []error { return m }

// OrderItemValidationError is the validation error returned by
// OrderItem.Validate if the designated constraints aren't met.
type OrderItemValidationError struct {
	field  string
	reason string
	cause  error
	key    bool
}

// Field function returns field value.
func (e OrderItemValidationError) Field() string { return e.field }

// Reason function returns reason value.
func (e OrderItemValidationError) Reason() string { return e.reason }

// Cause function returns cause value.
func (e OrderItemValidationError) Cause() error { return e.cause }

// Key function returns key value.
func (e OrderItemValidationError) Key() bool { return e.key }

// ErrorName returns error name.
func (e OrderItemValidationError) ErrorName() string { return "OrderItemValidationError" }

// Error satisfies the builtin error interface
func (e OrderItemValidationError) Error() string {
	cause := ""
	if e.cause != nil {
		cause = fmt.Sprintf(" | caused by: %v", e.cause)
	}

	key := ""
	if e.key {
		key = "key for "
	}

	return fmt.Sprintf(
		"invalid %sOrderItem.%s: %s%s",
		key,
		e.field,
		e.reason,
		cause)
}

var _ error = OrderItemValidationError{}

var _ interface {
	Field() string
	Reason() string
	Key() bool
	Cause() error
	ErrorName() string
} = OrderItemValidationError{}

// Validate checks the field values on OrderResult with the rules defined in
// the proto definition for this message. If any rules are violated, the first
// error encountered is returned, or nil if there are no violations.
func (m *OrderResult) Validate() error {
	return m.validate(false)
}

// ValidateAll checks the field values on OrderResult with the rules defined in
// the proto definition for this message. If any rules are violated, the
// result is a list of violation errors wrapped in OrderResultMultiError, or
// nil if none found.
func (m *OrderResult) ValidateAll() error {
	return m.validate(true)
}

func (m *OrderResult) validate(all bool) error {
	if m == nil {
		return nil
	}

	var errors []error

	// no validation rules for OrderId

	if len(errors) > 0 {
		return OrderResultMultiError(errors)
	}

	return nil
}

// OrderResultMultiError is an error wrapping multiple validation errors
// returned by OrderResult.ValidateAll() if the designated constraints aren't met.
type OrderResultMultiError []error

// Error returns a concatenation of all the error messages it wraps.
func (m OrderResultMultiError) Error() string {
	msgs := make([]string, 0, len(m))
	for _, err := range m {
		msgs = append(msgs, err.Error())
	}
	return strings.Join(msgs, "; ")
}

// AllErrors returns a list of validation violation errors.
func (m OrderResultMultiError) AllErrors() []error { return m }

// OrderResultValidationError is the validation error returned by
// OrderResult.Validate if the designated constraints aren't met.
type OrderResultValidationError struct {
	field  string
	reason string
	cause  error
	key    bool
}

// Field function returns field value.
func (e OrderResultValidationError) Field() string { return e.field }

// Reason function returns reason value.
func (e OrderResultValidationError) Reason() string { return e.reason }

// Cause function returns cause value.
func (e OrderResultValidationError) Cause() error { return e.cause }

// Key function returns key value.
func (e OrderResultValidationError) Key() bool { return e.key }

// ErrorName returns error name.
func (e OrderResultValidationError) ErrorName() string { return "OrderResultValidationError" }

// Error satisfies the builtin error interface
func (e OrderResultValidationError) Error() string {
	cause := ""
	if e.cause != nil {
		cause = fmt.Sprintf(" | caused by: %v", e.cause)
	}

	key := ""
	if e.key {
		key = "key for "
	}

	return fmt.Sprintf(
		"invalid %sOrderResult.%s: %s%s",
		key,
		e.field,
		e.reason,
		cause)
}

var _ error = OrderResultValidationError{}

var _ interface {
	Field() string
	Reason() string
	Key() bool
	Cause() error
	ErrorName() string
} = OrderResultValidationError{}

// Validate checks the field values on PlaceOrderResp with the rules defined in
// the proto definition for this message. If any rules are violated, the first
// error encountered is returned, or nil if there are no violations.
func (m *PlaceOrderResp) Validate() error {
	return m.validate(false)
}

// ValidateAll checks the field values on PlaceOrderResp with the rules defined
// in the proto definition for this message. If any rules are violated, the
// result is a list of violation errors wrapped in PlaceOrderRespMultiError,
// or nil if none found.
func (m *PlaceOrderResp) ValidateAll() error {
	return m.validate(true)
}

func (m *PlaceOrderResp) validate(all bool) error {
	if m == nil {
		return nil
	}

	var errors []error

	if all {
		switch v := interface{}(m.GetOrder()).(type) {
		case interface{ ValidateAll() error }:
			if err := v.ValidateAll(); err != nil {
				errors = append(errors, PlaceOrderRespValidationError{
					field:  "Order",
					reason: "embedded message failed validation",
					cause:  err,
				})
			}
		case interface{ Validate() error }:
			if err := v.Validate(); err != nil {
				errors = append(errors, PlaceOrderRespValidationError{
					field:  "Order",
					reason: "embedded message failed validation",
					cause:  err,
				})
			}
		}
	} else if v, ok := interface{}(m.GetOrder()).(interface{ Validate() error }); ok {
		if err := v.Validate(); err != nil {
			return PlaceOrderRespValidationError{
				field:  "Order",
				reason: "embedded message failed validation",
				cause:  err,
			}
		}
	}

	if len(errors) > 0 {
		return PlaceOrderRespMultiError(errors)
	}

	return nil
}

// PlaceOrderRespMultiError is an error wrapping multiple validation errors
// returned by PlaceOrderResp.ValidateAll() if the designated constraints
// aren't met.
type PlaceOrderRespMultiError []error

// Error returns a concatenation of all the error messages it wraps.
func (m PlaceOrderRespMultiError) Error() string {
	msgs := make([]string, 0, len(m))
	for _, err := range m {
		msgs = append(msgs, err.Error())
	}
	return strings.Join(msgs, "; ")
}

// AllErrors returns a list of validation violation errors.
func (m PlaceOrderRespMultiError) AllErrors() []error { return m }

// PlaceOrderRespValidationError is the validation error returned by
// PlaceOrderResp.Validate if the designated constraints aren't met.
type PlaceOrderRespValidationError struct {
	field  string
	reason string
	cause  error
	key    bool
}

// Field function returns field value.
func (e PlaceOrderRespValidationError) Field() string { return e.field }

// Reason function returns reason value.
func (e PlaceOrderRespValidationError) Reason() string { return e.reason }

// Cause function returns cause value.
func (e PlaceOrderRespValidationError) Cause() error { return e.cause }

// Key function returns key value.
func (e PlaceOrderRespValidationError) Key() bool { return e.key }

// ErrorName returns error name.
func (e PlaceOrderRespValidationError) ErrorName() string { return "PlaceOrderRespValidationError" }

// Error satisfies the builtin error interface
func (e PlaceOrderRespValidationError) Error() string {
	cause := ""
	if e.cause != nil {
		cause = fmt.Sprintf(" | caused by: %v", e.cause)
	}

	key := ""
	if e.key {
		key = "key for "
	}

	return fmt.Sprintf(
		"invalid %sPlaceOrderResp.%s: %s%s",
		key,
		e.field,
		e.reason,
		cause)
}

var _ error = PlaceOrderRespValidationError{}

var _ interface {
	Field() string
	Reason() string
	Key() bool
	Cause() error
	ErrorName() string
} = PlaceOrderRespValidationError{}

// Validate checks the field values on Order with the rules defined in the
// proto definition for this message. If any rules are violated, the first
// error encountered is returned, or nil if there are no violations.
func (m *Order) Validate() error {
	return m.validate(false)
}

// ValidateAll checks the field values on Order with the rules defined in the
// proto definition for this message. If any rules are violated, the result is
// a list of violation errors wrapped in OrderMultiError, or nil if none found.
func (m *Order) ValidateAll() error {
	return m.validate(true)
}

func (m *Order) validate(all bool) error {
	if m == nil {
		return nil
	}

	var errors []error

	for idx, item := range m.GetItems() {
		_, _ = idx, item

		if all {
			switch v := interface{}(item).(type) {
			case interface{ ValidateAll() error }:
				if err := v.ValidateAll(); err != nil {
					errors = append(errors, OrderValidationError{
						field:  fmt.Sprintf("Items[%v]", idx),
						reason: "embedded message failed validation",
						cause:  err,
					})
				}
			case interface{ Validate() error }:
				if err := v.Validate(); err != nil {
					errors = append(errors, OrderValidationError{
						field:  fmt.Sprintf("Items[%v]", idx),
						reason: "embedded message failed validation",
						cause:  err,
					})
				}
			}
		} else if v, ok := interface{}(item).(interface{ Validate() error }); ok {
			if err := v.Validate(); err != nil {
				return OrderValidationError{
					field:  fmt.Sprintf("Items[%v]", idx),
					reason: "embedded message failed validation",
					cause:  err,
				}
			}
		}

	}

	// no validation rules for OrderId

	if utf8.RuneCountInString(m.GetUserId()) != 32 {
		err := OrderValidationError{
			field:  "UserId",
			reason: "value length must be 32 runes",
		}
		if !all {
			return err
		}
		errors = append(errors, err)

	}

	if err := m._validateUuid(m.GetUserId()); err != nil {
		err = OrderValidationError{
			field:  "UserId",
			reason: "value must be a valid UUID",
			cause:  err,
		}
		if !all {
			return err
		}
		errors = append(errors, err)
	}

	if utf8.RuneCountInString(m.GetCurrency()) != 3 {
		err := OrderValidationError{
			field:  "Currency",
			reason: "value length must be 3 runes",
		}
		if !all {
			return err
		}
		errors = append(errors, err)

	}

	if all {
		switch v := interface{}(m.GetAddress()).(type) {
		case interface{ ValidateAll() error }:
			if err := v.ValidateAll(); err != nil {
				errors = append(errors, OrderValidationError{
					field:  "Address",
					reason: "embedded message failed validation",
					cause:  err,
				})
			}
		case interface{ Validate() error }:
			if err := v.Validate(); err != nil {
				errors = append(errors, OrderValidationError{
					field:  "Address",
					reason: "embedded message failed validation",
					cause:  err,
				})
			}
		}
	} else if v, ok := interface{}(m.GetAddress()).(interface{ Validate() error }); ok {
		if err := v.Validate(); err != nil {
			return OrderValidationError{
				field:  "Address",
				reason: "embedded message failed validation",
				cause:  err,
			}
		}
	}

	// no validation rules for Email

	if all {
		switch v := interface{}(m.GetCreatedAt()).(type) {
		case interface{ ValidateAll() error }:
			if err := v.ValidateAll(); err != nil {
				errors = append(errors, OrderValidationError{
					field:  "CreatedAt",
					reason: "embedded message failed validation",
					cause:  err,
				})
			}
		case interface{ Validate() error }:
			if err := v.Validate(); err != nil {
				errors = append(errors, OrderValidationError{
					field:  "CreatedAt",
					reason: "embedded message failed validation",
					cause:  err,
				})
			}
		}
	} else if v, ok := interface{}(m.GetCreatedAt()).(interface{ Validate() error }); ok {
		if err := v.Validate(); err != nil {
			return OrderValidationError{
				field:  "CreatedAt",
				reason: "embedded message failed validation",
				cause:  err,
			}
		}
	}

	// no validation rules for PaymentStatus

	if len(errors) > 0 {
		return OrderMultiError(errors)
	}

	return nil
}

func (m *Order) _validateUuid(uuid string) error {
	if matched := _order_uuidPattern.MatchString(uuid); !matched {
		return errors.New("invalid uuid format")
	}

	return nil
}

// OrderMultiError is an error wrapping multiple validation errors returned by
// Order.ValidateAll() if the designated constraints aren't met.
type OrderMultiError []error

// Error returns a concatenation of all the error messages it wraps.
func (m OrderMultiError) Error() string {
	msgs := make([]string, 0, len(m))
	for _, err := range m {
		msgs = append(msgs, err.Error())
	}
	return strings.Join(msgs, "; ")
}

// AllErrors returns a list of validation violation errors.
func (m OrderMultiError) AllErrors() []error { return m }

// OrderValidationError is the validation error returned by Order.Validate if
// the designated constraints aren't met.
type OrderValidationError struct {
	field  string
	reason string
	cause  error
	key    bool
}

// Field function returns field value.
func (e OrderValidationError) Field() string { return e.field }

// Reason function returns reason value.
func (e OrderValidationError) Reason() string { return e.reason }

// Cause function returns cause value.
func (e OrderValidationError) Cause() error { return e.cause }

// Key function returns key value.
func (e OrderValidationError) Key() bool { return e.key }

// ErrorName returns error name.
func (e OrderValidationError) ErrorName() string { return "OrderValidationError" }

// Error satisfies the builtin error interface
func (e OrderValidationError) Error() string {
	cause := ""
	if e.cause != nil {
		cause = fmt.Sprintf(" | caused by: %v", e.cause)
	}

	key := ""
	if e.key {
		key = "key for "
	}

	return fmt.Sprintf(
		"invalid %sOrder.%s: %s%s",
		key,
		e.field,
		e.reason,
		cause)
}

var _ error = OrderValidationError{}

var _ interface {
	Field() string
	Reason() string
	Key() bool
	Cause() error
	ErrorName() string
} = OrderValidationError{}

// Validate checks the field values on GetConsumerOrdersReq with the rules
// defined in the proto definition for this message. If any rules are
// violated, the first error encountered is returned, or nil if there are no violations.
func (m *GetConsumerOrdersReq) Validate() error {
	return m.validate(false)
}

// ValidateAll checks the field values on GetConsumerOrdersReq with the rules
// defined in the proto definition for this message. If any rules are
// violated, the result is a list of violation errors wrapped in
// GetConsumerOrdersReqMultiError, or nil if none found.
func (m *GetConsumerOrdersReq) ValidateAll() error {
	return m.validate(true)
}

func (m *GetConsumerOrdersReq) validate(all bool) error {
	if m == nil {
		return nil
	}

	var errors []error

	// no validation rules for UserId

	// no validation rules for Page

	// no validation rules for PageSize

	if len(errors) > 0 {
		return GetConsumerOrdersReqMultiError(errors)
	}

	return nil
}

// GetConsumerOrdersReqMultiError is an error wrapping multiple validation
// errors returned by GetConsumerOrdersReq.ValidateAll() if the designated
// constraints aren't met.
type GetConsumerOrdersReqMultiError []error

// Error returns a concatenation of all the error messages it wraps.
func (m GetConsumerOrdersReqMultiError) Error() string {
	msgs := make([]string, 0, len(m))
	for _, err := range m {
		msgs = append(msgs, err.Error())
	}
	return strings.Join(msgs, "; ")
}

// AllErrors returns a list of validation violation errors.
func (m GetConsumerOrdersReqMultiError) AllErrors() []error { return m }

// GetConsumerOrdersReqValidationError is the validation error returned by
// GetConsumerOrdersReq.Validate if the designated constraints aren't met.
type GetConsumerOrdersReqValidationError struct {
	field  string
	reason string
	cause  error
	key    bool
}

// Field function returns field value.
func (e GetConsumerOrdersReqValidationError) Field() string { return e.field }

// Reason function returns reason value.
func (e GetConsumerOrdersReqValidationError) Reason() string { return e.reason }

// Cause function returns cause value.
func (e GetConsumerOrdersReqValidationError) Cause() error { return e.cause }

// Key function returns key value.
func (e GetConsumerOrdersReqValidationError) Key() bool { return e.key }

// ErrorName returns error name.
func (e GetConsumerOrdersReqValidationError) ErrorName() string {
	return "GetConsumerOrdersReqValidationError"
}

// Error satisfies the builtin error interface
func (e GetConsumerOrdersReqValidationError) Error() string {
	cause := ""
	if e.cause != nil {
		cause = fmt.Sprintf(" | caused by: %v", e.cause)
	}

	key := ""
	if e.key {
		key = "key for "
	}

	return fmt.Sprintf(
		"invalid %sGetConsumerOrdersReq.%s: %s%s",
		key,
		e.field,
		e.reason,
		cause)
}

var _ error = GetConsumerOrdersReqValidationError{}

var _ interface {
	Field() string
	Reason() string
	Key() bool
	Cause() error
	ErrorName() string
} = GetConsumerOrdersReqValidationError{}

// Validate checks the field values on GetAllOrdersReq with the rules defined
// in the proto definition for this message. If any rules are violated, the
// first error encountered is returned, or nil if there are no violations.
func (m *GetAllOrdersReq) Validate() error {
	return m.validate(false)
}

// ValidateAll checks the field values on GetAllOrdersReq with the rules
// defined in the proto definition for this message. If any rules are
// violated, the result is a list of violation errors wrapped in
// GetAllOrdersReqMultiError, or nil if none found.
func (m *GetAllOrdersReq) ValidateAll() error {
	return m.validate(true)
}

func (m *GetAllOrdersReq) validate(all bool) error {
	if m == nil {
		return nil
	}

	var errors []error

	// no validation rules for Page

	// no validation rules for PageSize

	if len(errors) > 0 {
		return GetAllOrdersReqMultiError(errors)
	}

	return nil
}

// GetAllOrdersReqMultiError is an error wrapping multiple validation errors
// returned by GetAllOrdersReq.ValidateAll() if the designated constraints
// aren't met.
type GetAllOrdersReqMultiError []error

// Error returns a concatenation of all the error messages it wraps.
func (m GetAllOrdersReqMultiError) Error() string {
	msgs := make([]string, 0, len(m))
	for _, err := range m {
		msgs = append(msgs, err.Error())
	}
	return strings.Join(msgs, "; ")
}

// AllErrors returns a list of validation violation errors.
func (m GetAllOrdersReqMultiError) AllErrors() []error { return m }

// GetAllOrdersReqValidationError is the validation error returned by
// GetAllOrdersReq.Validate if the designated constraints aren't met.
type GetAllOrdersReqValidationError struct {
	field  string
	reason string
	cause  error
	key    bool
}

// Field function returns field value.
func (e GetAllOrdersReqValidationError) Field() string { return e.field }

// Reason function returns reason value.
func (e GetAllOrdersReqValidationError) Reason() string { return e.reason }

// Cause function returns cause value.
func (e GetAllOrdersReqValidationError) Cause() error { return e.cause }

// Key function returns key value.
func (e GetAllOrdersReqValidationError) Key() bool { return e.key }

// ErrorName returns error name.
func (e GetAllOrdersReqValidationError) ErrorName() string { return "GetAllOrdersReqValidationError" }

// Error satisfies the builtin error interface
func (e GetAllOrdersReqValidationError) Error() string {
	cause := ""
	if e.cause != nil {
		cause = fmt.Sprintf(" | caused by: %v", e.cause)
	}

	key := ""
	if e.key {
		key = "key for "
	}

	return fmt.Sprintf(
		"invalid %sGetAllOrdersReq.%s: %s%s",
		key,
		e.field,
		e.reason,
		cause)
}

var _ error = GetAllOrdersReqValidationError{}

var _ interface {
	Field() string
	Reason() string
	Key() bool
	Cause() error
	ErrorName() string
} = GetAllOrdersReqValidationError{}

// Validate checks the field values on Orders with the rules defined in the
// proto definition for this message. If any rules are violated, the first
// error encountered is returned, or nil if there are no violations.
func (m *Orders) Validate() error {
	return m.validate(false)
}

// ValidateAll checks the field values on Orders with the rules defined in the
// proto definition for this message. If any rules are violated, the result is
// a list of violation errors wrapped in OrdersMultiError, or nil if none found.
func (m *Orders) ValidateAll() error {
	return m.validate(true)
}

func (m *Orders) validate(all bool) error {
	if m == nil {
		return nil
	}

	var errors []error

	for idx, item := range m.GetOrders() {
		_, _ = idx, item

		if all {
			switch v := interface{}(item).(type) {
			case interface{ ValidateAll() error }:
				if err := v.ValidateAll(); err != nil {
					errors = append(errors, OrdersValidationError{
						field:  fmt.Sprintf("Orders[%v]", idx),
						reason: "embedded message failed validation",
						cause:  err,
					})
				}
			case interface{ Validate() error }:
				if err := v.Validate(); err != nil {
					errors = append(errors, OrdersValidationError{
						field:  fmt.Sprintf("Orders[%v]", idx),
						reason: "embedded message failed validation",
						cause:  err,
					})
				}
			}
		} else if v, ok := interface{}(item).(interface{ Validate() error }); ok {
			if err := v.Validate(); err != nil {
				return OrdersValidationError{
					field:  fmt.Sprintf("Orders[%v]", idx),
					reason: "embedded message failed validation",
					cause:  err,
				}
			}
		}

	}

	if len(errors) > 0 {
		return OrdersMultiError(errors)
	}

	return nil
}

// OrdersMultiError is an error wrapping multiple validation errors returned by
// Orders.ValidateAll() if the designated constraints aren't met.
type OrdersMultiError []error

// Error returns a concatenation of all the error messages it wraps.
func (m OrdersMultiError) Error() string {
	msgs := make([]string, 0, len(m))
	for _, err := range m {
		msgs = append(msgs, err.Error())
	}
	return strings.Join(msgs, "; ")
}

// AllErrors returns a list of validation violation errors.
func (m OrdersMultiError) AllErrors() []error { return m }

// OrdersValidationError is the validation error returned by Orders.Validate if
// the designated constraints aren't met.
type OrdersValidationError struct {
	field  string
	reason string
	cause  error
	key    bool
}

// Field function returns field value.
func (e OrdersValidationError) Field() string { return e.field }

// Reason function returns reason value.
func (e OrdersValidationError) Reason() string { return e.reason }

// Cause function returns cause value.
func (e OrdersValidationError) Cause() error { return e.cause }

// Key function returns key value.
func (e OrdersValidationError) Key() bool { return e.key }

// ErrorName returns error name.
func (e OrdersValidationError) ErrorName() string { return "OrdersValidationError" }

// Error satisfies the builtin error interface
func (e OrdersValidationError) Error() string {
	cause := ""
	if e.cause != nil {
		cause = fmt.Sprintf(" | caused by: %v", e.cause)
	}

	key := ""
	if e.key {
		key = "key for "
	}

	return fmt.Sprintf(
		"invalid %sOrders.%s: %s%s",
		key,
		e.field,
		e.reason,
		cause)
}

var _ error = OrdersValidationError{}

var _ interface {
	Field() string
	Reason() string
	Key() bool
	Cause() error
	ErrorName() string
} = OrdersValidationError{}

// Validate checks the field values on MarkOrderPaidReq with the rules defined
// in the proto definition for this message. If any rules are violated, the
// first error encountered is returned, or nil if there are no violations.
func (m *MarkOrderPaidReq) Validate() error {
	return m.validate(false)
}

// ValidateAll checks the field values on MarkOrderPaidReq with the rules
// defined in the proto definition for this message. If any rules are
// violated, the result is a list of violation errors wrapped in
// MarkOrderPaidReqMultiError, or nil if none found.
func (m *MarkOrderPaidReq) ValidateAll() error {
	return m.validate(true)
}

func (m *MarkOrderPaidReq) validate(all bool) error {
	if m == nil {
		return nil
	}

	var errors []error

	// no validation rules for OrderId

	if len(errors) > 0 {
		return MarkOrderPaidReqMultiError(errors)
	}

	return nil
}

// MarkOrderPaidReqMultiError is an error wrapping multiple validation errors
// returned by MarkOrderPaidReq.ValidateAll() if the designated constraints
// aren't met.
type MarkOrderPaidReqMultiError []error

// Error returns a concatenation of all the error messages it wraps.
func (m MarkOrderPaidReqMultiError) Error() string {
	msgs := make([]string, 0, len(m))
	for _, err := range m {
		msgs = append(msgs, err.Error())
	}
	return strings.Join(msgs, "; ")
}

// AllErrors returns a list of validation violation errors.
func (m MarkOrderPaidReqMultiError) AllErrors() []error { return m }

// MarkOrderPaidReqValidationError is the validation error returned by
// MarkOrderPaidReq.Validate if the designated constraints aren't met.
type MarkOrderPaidReqValidationError struct {
	field  string
	reason string
	cause  error
	key    bool
}

// Field function returns field value.
func (e MarkOrderPaidReqValidationError) Field() string { return e.field }

// Reason function returns reason value.
func (e MarkOrderPaidReqValidationError) Reason() string { return e.reason }

// Cause function returns cause value.
func (e MarkOrderPaidReqValidationError) Cause() error { return e.cause }

// Key function returns key value.
func (e MarkOrderPaidReqValidationError) Key() bool { return e.key }

// ErrorName returns error name.
func (e MarkOrderPaidReqValidationError) ErrorName() string { return "MarkOrderPaidReqValidationError" }

// Error satisfies the builtin error interface
func (e MarkOrderPaidReqValidationError) Error() string {
	cause := ""
	if e.cause != nil {
		cause = fmt.Sprintf(" | caused by: %v", e.cause)
	}

	key := ""
	if e.key {
		key = "key for "
	}

	return fmt.Sprintf(
		"invalid %sMarkOrderPaidReq.%s: %s%s",
		key,
		e.field,
		e.reason,
		cause)
}

var _ error = MarkOrderPaidReqValidationError{}

var _ interface {
	Field() string
	Reason() string
	Key() bool
	Cause() error
	ErrorName() string
} = MarkOrderPaidReqValidationError{}

// Validate checks the field values on MarkOrderPaidResp with the rules defined
// in the proto definition for this message. If any rules are violated, the
// first error encountered is returned, or nil if there are no violations.
func (m *MarkOrderPaidResp) Validate() error {
	return m.validate(false)
}

// ValidateAll checks the field values on MarkOrderPaidResp with the rules
// defined in the proto definition for this message. If any rules are
// violated, the result is a list of violation errors wrapped in
// MarkOrderPaidRespMultiError, or nil if none found.
func (m *MarkOrderPaidResp) ValidateAll() error {
	return m.validate(true)
}

func (m *MarkOrderPaidResp) validate(all bool) error {
	if m == nil {
		return nil
	}

	var errors []error

	if len(errors) > 0 {
		return MarkOrderPaidRespMultiError(errors)
	}

	return nil
}

// MarkOrderPaidRespMultiError is an error wrapping multiple validation errors
// returned by MarkOrderPaidResp.ValidateAll() if the designated constraints
// aren't met.
type MarkOrderPaidRespMultiError []error

// Error returns a concatenation of all the error messages it wraps.
func (m MarkOrderPaidRespMultiError) Error() string {
	msgs := make([]string, 0, len(m))
	for _, err := range m {
		msgs = append(msgs, err.Error())
	}
	return strings.Join(msgs, "; ")
}

// AllErrors returns a list of validation violation errors.
func (m MarkOrderPaidRespMultiError) AllErrors() []error { return m }

// MarkOrderPaidRespValidationError is the validation error returned by
// MarkOrderPaidResp.Validate if the designated constraints aren't met.
type MarkOrderPaidRespValidationError struct {
	field  string
	reason string
	cause  error
	key    bool
}

// Field function returns field value.
func (e MarkOrderPaidRespValidationError) Field() string { return e.field }

// Reason function returns reason value.
func (e MarkOrderPaidRespValidationError) Reason() string { return e.reason }

// Cause function returns cause value.
func (e MarkOrderPaidRespValidationError) Cause() error { return e.cause }

// Key function returns key value.
func (e MarkOrderPaidRespValidationError) Key() bool { return e.key }

// ErrorName returns error name.
func (e MarkOrderPaidRespValidationError) ErrorName() string {
	return "MarkOrderPaidRespValidationError"
}

// Error satisfies the builtin error interface
func (e MarkOrderPaidRespValidationError) Error() string {
	cause := ""
	if e.cause != nil {
		cause = fmt.Sprintf(" | caused by: %v", e.cause)
	}

	key := ""
	if e.key {
		key = "key for "
	}

	return fmt.Sprintf(
		"invalid %sMarkOrderPaidResp.%s: %s%s",
		key,
		e.field,
		e.reason,
		cause)
}

var _ error = MarkOrderPaidRespValidationError{}

var _ interface {
	Field() string
	Reason() string
	Key() bool
	Cause() error
	ErrorName() string
} = MarkOrderPaidRespValidationError{}
